src/store/actions.js

```diff
@@ -502,30 +502,19 @@ export default {
 			newUid: uid,
 		})
 	},
-	deleteMessage({getters, commit}, envelope) {
-		const folder = getters.getFolder(envelope.accountId, envelope.folderId)
-		commit('removeEnvelope', {
-			accountId: envelope.accountId,
-			folderId: envelope.folderId,
-			id: envelope.id,
-		})
+	deleteMessage({getters, commit}, accountId, folderId, id) {
+		commit('removeEnvelope', {accountId, folderId, id})
 
-		return deleteMessage(envelope.accountId, envelope.folderId, envelope.id)
+		return deleteMessage(accountId, folderId, id)
 			.then(() => {
-				commit('removeMessage', {
-					accountId: envelope.accountId,
-					folder,
-					id: envelope.id,
-				})
+				const folder = getters.getFolder(accountId, folderId)
+				commit('removeMessage', {accountId, folder, id})
 				console.log('message removed')
 			})
 			.catch(err => {
 				console.error('could not delete message', err)
-				commit('addEnvelope', {
-					accountId: envelope.accountId,
-					folderId: envelope.folderId,
-					envelope,
-				})
+				const envelope = getters.getEnvelope(accountId, folderId, id)
+				commit('addEnvelope', {accountId, folderId, envelope})
```

Comment:
Similar here, add an `undefined` check and if we don't have the envelope locally we don't need to commit any change